PhotoFeed/app/src/main/java/com/example/yhuang/photofeed/Database/PhotoHelper.java

```diff
@@ -0,0 +1,98 @@
+package com.example.yhuang.photofeed.Database;
+
+import android.content.ContentValues;
+import android.content.Context;
+import android.database.Cursor;
+import android.database.sqlite.SQLiteDatabase;
+import android.database.sqlite.SQLiteOpenHelper;
+
+import org.apache.commons.logging.Log;
+
+import java.util.ArrayList;
+
+/**
+ * Created by yhuang on 9/17/2015.
+ */
+public class PhotoHelper extends SQLiteOpenHelper {
+    private static final int DATABASE_VERSION = 2;
+    static final String DATABASE_NAME = "photo.db";
+
+    public PhotoHelper(Context context) {
+        super(context, DATABASE_NAME, null, DATABASE_VERSION);
+    }
+
+    @Override
+    public void onCreate(SQLiteDatabase db) {
+        db.execSQL(PhotoContract.SQL_CREATE_ENTRIES);
+    }
+
+    @Override
+    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
+        db.execSQL(PhotoContract.SQL_DELETE_ENTRIES);
+        onCreate(db);
+    }
+
+    public static void SyncWithDatabase(PhotoHelper mDbHelper, ArrayList<String> items) {
+        SQLiteDatabase db = mDbHelper.getWritableDatabase();
```

Comment:
You will see a performance increase if you move this line into the oncreate and only instantiate the writable database once, instead of every time you call sync.